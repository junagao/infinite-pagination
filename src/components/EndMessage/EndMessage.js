import React from 'react'
import Text from './EndMessage.styles'

export default function EndMessage() {
  return (
    <Text data-testid="end-message">
      <span role="img" aria-label="stop hand emoticon">
        âœ‹
      </span>{' '}
      You have reached the end.{' '}
    </Text>
  )
}
