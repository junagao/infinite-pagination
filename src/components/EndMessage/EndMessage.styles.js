import styled from '@emotion/styled'

const Text = styled.p`
  font-weight: 400;
`

export default Text
