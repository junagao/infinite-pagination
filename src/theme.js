const theme = {
  colors: {
    background: '#fff',
    primary: '#111',
    text: { body: '#111', hover: '#f0f0f0' },
    disabled: '#ddd',
    error: '#BD271E',
  },
  fontSizes: {
    xl: '4rem',
    lg: '2rem',
    md: '1.5rem',
    sm: '1rem',
    xs: '0.75rem',
  },
  space: [
    0,
    '0.25rem',
    '0.5rem',
    '0.75rem',
    '1rem',
    '1.5rem',
    '2rem',
    '4rem',
    '6rem',
  ],
  borders: {
    none: 'none',
    thin: '1px solid',
  },
  radii: {
    none: 0,
    base: '0.25em',
    round: '99999em',
  },
}

export default theme
