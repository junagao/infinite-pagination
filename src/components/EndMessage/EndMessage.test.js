import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'emotion-theming'
import { matchers } from 'jest-emotion'
import theme from '../../theme'
import EndMessage from './EndMessage'

expect.extend(matchers)

describe('EndMessage', () => {
  test('matches snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <EndMessage />
      </ThemeProvider>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('displays the correct end message', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <EndMessage />
      </ThemeProvider>,
    )
    expect(getByTestId('end-message')).toHaveTextContent(
      'âœ‹ You have reached the end.',
    )
  })

  test('renders the end message with the correct styles from theme', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <EndMessage />
      </ThemeProvider>,
    )
    expect(getByTestId('end-message')).toHaveStyleRule('font-size', '1.5rem')
    expect(getByTestId('end-message')).toHaveStyleRule('color', '#111')
    expect(getByTestId('end-message')).toHaveStyleRule('padding', '2rem 0')
  })
})
