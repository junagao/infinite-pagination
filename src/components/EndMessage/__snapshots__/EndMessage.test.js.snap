// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndMessage matches snapshot 1`] = `
.emotion-0 {
  font-size: 1.5rem;
  color: #111;
  padding: 2rem 0;
}

<p
  class="emotion-0 emotion-1"
  data-testid="end-message"
>
  <span
    aria-label="stop hand emoticon"
    role="img"
  >
    âœ‹
  </span>
   
  You have reached the end.
   
</p>
`;
